python/plugins/db_manager/db_plugins/data_model.py

```diff
@@ -139,6 +148,43 @@ def rowCount(self, index=None):
         return self.table.rowCount if self.table.rowCount is not None and self.columnCount(index) > 0 else 0
 
 
+class SqlResultModelAsync(QObject):
+
+    done = pyqtSignal()
+
+    def __init__(self):
+        QObject.__init__(self)
+        self.error = BaseError('')
+        self.status = None
+        self.model = None
+        self.task = None
+        self.canceled = False
+
+    def cancel(self):
+        self.canceled = True
+        if self.task:
+            self.task.cancel()
+
+    def modelDone(self):
+        if self.task:
+            self.status = self.task.status
+            self.model = self.task.model
+            self.error = self.task.error
```

Comment:
Might be safest to clear self.task here too - it'll be deleted shortly after by the task manager

Reply:
Done